biblib/views/library_view.py

```diff
@@ -175,11 +174,11 @@ def solr_update_library(cls, library_id, solr_docs, session):
                     updates['duplicates_removed'] += 1
             else:
                 new_library_bibcodes[bibcode] = library.bibcode[bibcode]
-        session.add(library)
-        session.commit()
-        if updates['update_list']:                
-            cls.update_database(session, library, new_library_bibcodes, updates)
-
+        
+        library.bibcode = new_library_bibcodes
+        cls.update_library(session, library)
```

Comment:
This modification looks good. The only change I would make would be to add some sort of `update` flag that is set to `True` if any of the above things are changed. If we don't do that, I think `sqlalchemy_continuum` will capture a lot of duplicate library updates as we are explicitly marking the library as modified every time we pass it to the `update_library` method.

Reply:
I see what you mean. I just updated and put the flags back.